de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/dvlm/mat/MatImporterTest.java

```diff
@@ -0,0 +1,343 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.dvlm.mat;
+
+import static org.junit.Assert.assertEquals;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.common.util.URI;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.APropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.model.extension.tests.model.EReferenceTest;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryAllProperty;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryComposition;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryReference;
+import de.dlr.sc.virsat.model.extension.tests.model.TestStructuralElement;
+import de.dlr.sc.virsat.model.extension.tests.model.TestStructuralElementOther;
+import de.dlr.sc.virsat.model.extension.tests.test.ATestConceptTestCase;
+import de.dlr.sc.virsat.model.external.tests.ExternalTestType;
+import us.hebi.matlab.mat.types.MatFile;
+
+public class MatImporterTest extends ATestConceptTestCase {
+
+	private StructuralElementInstance sei;
+	private TestStructuralElement tsei;
+	private MatExporter exporter;
+	private MatImporter importer;
+	private MatFile mat;
+	private static final int NUMBEROFELEMENTS = 6;
+
+	@Before
+	public void setUp() throws CoreException {
+		exporter = new MatExporter();
+		importer = new MatImporter();
+		super.setUp();
+		addEditingDomainAndRepository();
+		executeAsCommand(() -> loadTestConcept());
+		tsei = new TestStructuralElement(testConcept);
+		sei = tsei.getStructuralElementInstance();
+		sei.setName("testsei");
+		executeAsCommand(() -> repository.getRootEntities().add(sei));
+		mat = exporter.exportSei(sei);

```

Comment:
Could we also have one test that doesn't use the exporter to test the importer? By creating this matlab-string manually for example? Because like this, if the exporter might have a bug, we would maybe also miss bugs in the importer test

Reply:
Nice 👍 Thanks :)